### ISSUE TITLE
Excessive delays in software SPI mode

### ISSUE THREAD

**Issue by JoeEarlam** [adafruit/Adafruit-MAX31855-library#27]
The default software SPI implementation uses delay(ms) during every loop iteration, causing a single read of both internal and t/c temperatures to take almost 120ms.

I've removed this delay call (and removed/changed a few others) at my end and the code still works, with a read now taking around 3ms (still slower than 125KHz SPI but not by much).

Is there a reason for these delay calls, or could they be removed to improve efficiency?

```
else 
{
  // software SPI
  digitalWrite(sclk, LOW);
  delayMicroseconds(10);

  for (i=31; i>=0; i--) 
  {
    digitalWrite(sclk, LOW);
    d <<= 1;
    if (digitalRead(miso)) 
    {
      d |= 1;
    }
    digitalWrite(sclk, HIGH);
  }
}
```

**Comment by tablatronix** [comment]
Yes! I was wondering what was going on, then looked at the code, it is 100 times slower than other libraries, and its just the delays

**Comment by caternuson** [comment]
Should be fixed by 1.0.5 release which moved to using BusIO SPIDevice.
https://github.com/adafruit/Adafruit-MAX31855-library/releases/tag/1.0.5
